### PR TITLE
Implement expiring cache based on scalacache

### PR DESCRIPTION
**Pull request by majk-p**
This PR introduces an improvement post https://github.com/ocadotechnology/sttp-oauth2/pull/298 providing more sophisticated caching backend, based on scalacache

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -130,18 +131,37 @@ lazy val `oauth2-cache` = crossProject(JSPlatform, JVMPlatform)
...
   .dependsOn(oauth2)
 
+// oauth2-cache-scalacache doesn't have JS support because scalacache doesn't compile for js https://github.com/cb372/scalacache/issues/354#issuecomment-913024231
+lazy val `oauth2-cache-scalacache` = project
+  .settings(
+    name := "sttp-oauth2-cache-scalacache",
+    libraryDependencies ++= Seq(
+      "com.github.cb372" %%% "scalacache-core" % Versions.scalaCache,
+      "com.github.cb372" %% "scalacache-caffeine" % Versions.scalaCache % Test,
+      "org.typelevel" %%% "cats-effect-kernel" % Versions.catsEffect,
```
Comment (matwojcik) [inline build.sbt:141]: Aren't kernel and std already dependency of `cats-effect`? Or even all above a dependency of `cats-effect-testkit`?

File: build.sbt [hunk 2]
```diff
@@ -130,18 +131,37 @@ lazy val `oauth2-cache` = crossProject(JSPlatform, JVMPlatform)
...
+    compilerPlugins
+  )
+  .dependsOn(`oauth2-cache`.jvm)
+
 // oauth2-cache-cats doesn't have JS support because cats effect does not provide realTimeInstant on JS
 lazy val `oauth2-cache-cats` = project
   .settings(
     name := "sttp-oauth2-cache-cats",
     libraryDependencies ++= Seq(
+      "com.github.cb372" %%% "scalacache-cats-effect" % "0.28.0",
```
Comment (matwojcik) [inline build.sbt:157]: why?

File: docs/caching.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+  def remove(key: K): F[Unit]
+}
+```
+
+As the user of the library you can either choose to implement your own cache mechanism, or go for one of the provided:
+
+| Class                     |Description                                                  | Import module     |
+|---------------------------|-------------------------------------------------------------|-------------------|
+| `CatsRefExpiringCache`    | Simple Cats Effect 3 Ref based implementation               | `"com.ocadotechnology" %% "sttp-oauth2-cache-cats" % "@VERSION@"` |
```
Comment (matwojcik) [inline docs/caching.md:24]:
```suggestion
| `CatsRefExpiringCache`    | Simple Cats Effect 3 Ref based implementation. Good enough for `CachingAccessTokenProvider`, but  for `CachingTokenIntrospection` it's recommended to use an instance which better handles memory (this instance does not periodically removes expired entries)             | `"com.ocadotechnology" %% "sttp-oauth2-cache-cats" % "@VERSION@"` |
```

File: build.sbt (hunk 1 above)
Comment (majk-p) [reply build.sbt:141]: Well they are, those are just the drawbacks of Copy-Paste's method :smile:

File: build.sbt (hunk 2 above)
Comment (majk-p) [reply build.sbt]: By mistake, removed

File: docs/caching.md (hunk 3 above)
Comment (majk-p) [reply docs/caching.md]: Fixed by hand to avoid typo in `does not periodically removes`

**Review by matwojcik — APPROVED** [review]

**Pull request merged**